src/screens/Exchange/Exchange/Exchange.js

```diff
@@ -166,32 +106,46 @@ function Exchange() {
       if (!isCancelled) fromInputRef.current?.focus();
     }, 650);
 
-    isLogV2AppEvents() && dispatch(logEventAction('v2_exchange_pair_selected'));
-
     return () => {
       isCancelled = true;
     };
-  }, [fromAsset, dispatch]);
+  }, [dispatch, fromAsset]);
+
+  React.useEffect(() => {
+    if (isLogV2AppEvents()) {
+      dispatch(logEventAction('v2_exchange_pair_selected'));
+    }
+  }, [dispatch, chain, fromAddress, toAddress]);
```

Comment:
I would replace addresses with fromAsset and toAsset and remove chain.

Reply:
Changed